### PR TITLE
instructions: add SELFBALANCE from EIP-1884

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: include/evmc/instructions.h [hunk 1]
```diff
@@ -78,6 +78,7 @@ enum evmc_opcode
     OP_NUMBER = 0x43,
     OP_DIFFICULTY = 0x44,
     OP_GASLIMIT = 0x45,
+    OP_SELFBALANCE = 0x47,
```
Comment (axic) [inline include/evmc/instructions.h]: It is really 0x47, because chainid is 0x46: https://ethereum-magicians.org/t/opcode-repricing/3024/23?u=axic

**Comment by axic** [comment]
Ah, its not ready, need to update tests.

**Comment by axic** [comment]
@chfast @gumb0 please review

File: test/unittests/test_instructions.cpp [hunk 2]
```diff
@@ -268,6 +272,9 @@ TEST(instructions, istanbul_hard_fork)
     EXPECT_EQ(p[OP_CHAINID].gas_cost, -1);
     EXPECT_EQ(in[OP_CHAINID], std::string{"CHAINID"});
     EXPECT_TRUE(pn[OP_CHAINID] == nullptr);
+    EXPECT_EQ(i[OP_BALANCE].gas_cost, 700);
+    EXPECT_EQ(i[OP_EXTCODEHASH].gas_cost, 700);
+    EXPECT_EQ(i[OP_SLOAD].gas_cost, 800);
```
Comment (gumb0) [inline test/unittests/test_instructions.cpp]: check for `SELFBALANCE` missing here?

Comment (axic) [reply test/unittests/test_instructions.cpp]: Fixed.

**Comment by axic** [comment]
@chfast ok to merge?

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning].
 
 - Support for Istanbul CHAINID opcode. `chain_id` added to `evmc_tx_context` struct.
   [[#375](https://github.com/ethereum/evmc/pull/375)]
+- Support for Istanbul EIP-1884.
```
Comment (chfast) [inline CHANGELOG.md]: Add EIP name.

Comment (axic) [reply CHANGELOG.md]: What do you mean?

Comment (axic) [reply CHANGELOG.md]: Updated both this and chainid.

**Comment by axic** [comment]
@chfast should we wait for @holiman ?

**Comment by chfast** [comment]
We don't have to.

**Pull request merged**